### PR TITLE
Added a prow job for automatic merging of upstream changes into 1.9 envoy release branch

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
The job itself is expected to fail, as there's a conflict in VERSION between 1.9 istio release branch and upstream 1.17

**Comment by dmitri-d** [comment]
ping @PiotrSikora: is this what you had in mind?

File: prow/config/jobs/envoy-1.9.yaml
Comment (PiotrSikora) [inline prow/config/jobs/envoy-1.9.yaml]: Please also add `tide/merge-method-merge` label (see: https://github.com/istio/test-infra/pull/3213) and make sure the PR is merged, not squashed.

File: prow/config/jobs/envoy-1.9.yaml
Comment (PiotrSikora) [inline prow/config/jobs/envoy-1.9.yaml:68]: I have no idea what will happen afterwards. Is this merge commit going to be pushed as-is and used in a PR, or will the Automator try to create another commit or bail? I'm looking at [`automator.sh`](https://github.com/istio/test-infra/blob/master/tools/automator/automator.sh), and I don't see any support for this kind of workflow.

**Comment by ericvn** [comment]
Is there a particular need to do a periodic update of Envoy in the actual release streams? In the past, this wasn't done, and I believe that the RMs would want to know about incoming Envoy changes. This might be a good topic for the T&R workgroup call.

**Comment by PiotrSikora** [comment]
This pulls updates from Envoy's stable release branches (i.e. only security, reliability and build fixes). We already pull those as part of the sync during security release, but it's done manually and not in an automated fashion.

I think could remove `auto-merge` flag to let Release Managers review those changes before merging them, if you think that would be helpful.

File: prow/config/jobs/envoy-1.9.yaml
Comment (dmitri-d) [reply prow/config/jobs/envoy-1.9.yaml:68]: This doesn't actually create a commit, just stages the changes for one; I'm not sure what will happen if there were conflicts during a merge; I'll take a closer look at automator.sh. Any chance we could test this on a dummy branch?

**Comment by brian-avery** [comment]
I'm ok with this if we have the release manager/networking approval on PRs. I'm not sure what we would pull in with automated updates, but they can decide if it should be merged. I also think it's better to keep more in sync with Envoy which this would help with.

File: prow/config/jobs/envoy-1.9.yaml
Comment (dmitri-d) [reply prow/config/jobs/envoy-1.9.yaml]: > make sure the PR is merged, not squashed.

Not sure how/where this is controlled, could you point me to it please?

**Comment by dmitri-d** [comment]
* removed auto-merge label

**Review by PiotrSikora — APPROVED** [review]

**Review by brian-avery — APPROVED** [review]

**Comment by ericvn** [comment]
This test continues to fail...

https://prow.istio.io/job-history/gs/istio-prow/logs/update-envoy_envoy_release-1.9_periodic

**Pull request merged**
